### PR TITLE
Drop Python 3.5 support

### PR DESCRIPTION
**Pull request by jgosmann**
**Motivation and context:**

Drop support for Python 3.5 because nengo-bones does not work with it anymore.

* Change string interpolations and string `format()` calls to f-strings which were introduced with Python 3.6. A few `format()` calls remain where an f-string either cannot be used or would be less legible.

**Interactions with other PRs:**

Fixes build for all other PRs. Other PRs should switch over to f-strings.

**How has this been tested?**

existing tests

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Where should a reviewer start?**

Changes to the .nengobones.yml might be a good place?

**Types of changes:**

- Maintenance

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [n/a] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

**Still to do:**

* [x] Double check that missing string interpolation arguments have been filled in correctly.

### CODE REVIEW

**Review by Seanny123 — APPROVED** [review]

File: .nengobones.yml [hunk 1]
```diff
@@ -12,26 +12,27 @@ license_rst: {}
 
 manifest_in:
   global-exclude:
-    - "*.swp"  # exclude VIM swp files
+    - "*.swp" # exclude VIM swp files
 
 travis_yml:
-  python: 3.7
+  python: 3.9
```
Comment (hunse) [inline .nengobones.yml:18]: We don't officially support Python 3.9 yet in Nengo, though that's not to say it might not all work fine.

**Review by hunse — APPROVED** [review]

File: .nengobones.yml (hunk 1 above)
Comment (tbekolay) [reply .nengobones.yml:18]: Unofficially, I'm currently using Python 3.9, but with 3.8 the issue was Windows specific so who knows.

**Comment by jgosmann** [comment]
Force pushed a few more changes:
* Added 16fb5a66ee33104554c92c781b2744e2fbc2f38e
* Amended b427ddbdd2a2fbaa9d6d486d0b972d76953ce369 to fix docs build
* Added commit to fix deprecated intersphinx links

File: .nengobones.yml (hunk 1 above)
Comment (jgosmann) [reply .nengobones.yml]: I think it doesn't hurt to run the tests on 3.9 too as long as they pass. I also think that the "Python 3.6+" formulation in the the readme is fine (NengoSPA is more platform independent than Nengo, so it is even more unlikely that a new Python release breaks things). Might consider taking the classifier out of the setup.py ... but that I would have to re-add it later.

**Pull request merged**